src/Xamarin.Android.Build.Tasks/Tasks/AsyncTask.cs

```diff
@@ -251,6 +255,18 @@ private void LogInternal<T> (Queue queue, Action<T> action, ManualResetEvent res
 			}
 		}
 
+		protected void Yield ()
+		{
+			if (YieldDuringToolExecution && BuildEngine is IBuildEngine3)
+				(BuildEngine as IBuildEngine3).Yield ();
```

Comment:
Convention: `as` could return `null`, so `(value as Foo).Bar()` could result in a `NullReferenceException`.

If we are *asserting* that the cast *must* succeed, use a cast:

    ((IBuildEngine3) BuildEngine).Yield ();

Otherwise, `?.` should be used so we don't throw a `NullReferenceException`:

    (BuildEngine as IBuildEngine3)?.Yield ();

The `Reacquire()` invocation below should also be fixed.